### PR TITLE
Always force dask arrays to float in missing.interp_func

### PR DESCRIPTION
**Pull request by Illviljan**
`scipy.interpolate.interp1d` always forces float so make sure that `da.blockwise` understands that as well.

- [x] Workaround for  #4770
- [x] Tests added
- [x] Passes `isort . && black . && mypy . && flake8`

### CODE REVIEW

File: xarray/core/missing.py [hunk 1]
```diff
@@ -734,6 +734,13 @@ def interp_func(var, x, new_x, method, kwargs):
         # if usefull, re-use localize for each chunk of new_x
         localize = (method in ["linear", "nearest"]) and (new_x[0].chunks is not None)
 
+        # scipy.interpolate.interp1d always forces to float.
```
Comment (dcherian) [inline xarray/core/missing.py:737]: don't we also support the numpy interpolator? Should we return dtype from `_get_interpolator` to account for both cases?

Comment (Illviljan) [reply xarray/core/missing.py:737]: Maybe, but since vectorizable_only is set to True always we cant get to the numpyinterpolator, atleast in interp_func. Does np.interp also force float?

Knowing the dtype is only relevant for dask arrays also which would mean unneccesary rows of code for non-dasks.

I also have some dreams of supporting the correct dtypes in dataset.interp which would probably mean overwriting that suggested dtype anyway.

Comment (dcherian) [reply xarray/core/missing.py:737]: > Does np.interp also force float?

Yes, according to the docstring, so my comment is not relevant. Thanks.

**Review by dcherian — APPROVED** [review]

**Review by andersy005 — APPROVED** [review]

**Comment by mathause** [comment]
thanks @Illviljan! Do I understand correctly - this does not resolve #4770 and we should leave it open?

**Comment by Illviljan** [comment]
@mathause In the ideal world I think we should get back the same dtype as it was before interpolation. 
If we handle that with the same performance then it can be closed. Might be tough to close it though because when I've simply tested using .astype() it unfortunately slows down too much. 

Maybe rewording the title a bit is in order or should we make a new one?
